dydx/client.py

```diff
@@ -0,0 +1,165 @@
+import random
+import requests
+import dydx.util as utils
+
+from web3 import Web3
+
+
+class Client(object):
+
+    TAKER_ACCOUNT_OWNER = '0x0000000000000000000000000000000000000000'
+    TAKER_ACCOUNT_NUMBER = '0'
+    BASE_API_URI = 'https://api.dydx.exchange/v1/dex/'
+
+    def __init__(self, private_key, public_address=None, account_number='0'):
+        if type(private_key) == str:
+            private_key = \
+                bytearray.fromhex(utils.strip_hex_prefix(private_key))
+        elif type(private_key) != bytes:
+            raise TypeError('private_key incorrect type')
+        self.web3 = Web3()
+        self.private_key = private_key
+        self.account_number = account_number
+        self.public_address = utils.private_key_to_public_address(private_key)
+        if public_address:
+            assert(self.public_address == public_address.lower())
```

Comment:
will this give a reasonable error to the user? this will probably be a common user error, to give a private key that doesn't match the address